core/src/main/java/org/apache/accumulo/core/metadata/TabletLocationState.java

```diff
@@ -36,7 +36,7 @@ public class TabletLocationState {
 
   public static class BadLocationStateException extends Exception {
     private static final long serialVersionUID = 1L;
-    private Text metadataTableEntry;
+    private transient final Text metadataTableEntry;
```

Comment:
It feels incorrect to make this transient. The entire point of storing this value is so that it can be retrieved by the `getEncodedEndRow` method. If it is transient and does lose its value upon serializing, then that method will always return null. Nothing in this class protects against that risk by detecting it. The method would happily return null.

At the very least, we could make sure we check that it is non-null upon construction, and then detect that it has been lost in and throw an IllegalStateException in getEncodedEndRow() if the value is null.

Reply:
I don't know if it is, or will ever be, serialized. But the exceptions are supposed to always be serializable, because they very well could be at any time. Others may rely on that for this exception, and won't know to honor the assumption you made. Having an explicit detection of this case won't make serialization work... but it will prevent it from introducing bugs that are hard to find.

Reply:
Addressed in PR #2911